Tan_Heng_Yeow/progress.md

```diff
@@ -196,27 +200,80 @@ As mentioned in last week's retrospect, I intend to combine the work done for st
+
+I proceeded to implement a new column in the existing students' spreadsheet interface to show HTML green ticks as a form of experiment.
 
 Week 10 Plan:
 
 4th stage:
-- [ ] Finish up any remaining work from Week 9
-- [ ] Migrate status box messages to the new feedback mechanism.
+- [X] Finish up any remaining work from Week 9
+- [X] Migrate status box messages to the new feedback mechanism.
 - [ ] Write tests for new feature above
 
+Works:
+- Progress for this week marked at this [checkpoint](https://github.com/TEAMMATES/teammates/pull/8960/commits/d353ba96f328d497b958630464c124c04dd13b53).
+- Worked on this [PR](https://github.com/TEAMMATES/teammates/pull/8976).
+
+Retrospect on Week 10:
+
+Prior to this stage of the project, I created a new back-end action to perform mass updates of students as mentioned in Week 6's retrospect. The action would result in a page refresh after updating students and show success/failure indicators in the form of status messages. As pointed out in the Week 6's retrospect, this behavior is not desirable. This is because a page refresh would cause the existing students' spreadsheet to be empty. When the spreadsheet is expanded again upon user click, an AJAX action would be executed to fetch the latest copy of existing students' data. This results in complications showing users any immediate feedback of the data sent after backend validation. Upon further discussion with my project mentors, feedback for success cases should be shown as well, not only error cases.
+
+As such, the goal is to use an AJAX action to perform back-end validation of data and send the appropriate messages to the front-end. After which, the existing students' spreadsheet interface would be updated with the relevant feedback messages on the fly by modifying the DOM directly, without needing to refresh the page. 
+
+I did this change in incremental steps. I first created a new AJAX action to process updated students and return success/failure indicators. After that, I refactored both the front-end logic and the existing back-end action that performs mass update of students. As a result, the back-end action only does a page refresh upon a success indicator received from the front-end. In the final step, I replaced the back-end action completely with the current new AJAX action.
```

Comment:
this is not called "refactoring". "Refactor" means you do not change the external behaviour, just the internal mechanism. In this case, you are changing the external behaviour, isn't it?